app/trainer.py

```diff
@@ -0,0 +1,329 @@
+import json
+import logging
+import os
+from datetime import datetime
+from distutils.dir_util import copy_tree
+from typing import Any, Callable, List, Set
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from bert import tokenization
+from flask import Blueprint
+from models import class_based_attention
+
+import app.tasks as tasks
+from app.classifier import Classifier, ClassifierCache
+from app.embedder import MAX_SEQ_LENGTH, Embedder
+
+train_bp = Blueprint('train_bp', __name__)
+
+# LEARNING_RATE = 2e-3
+LEARNING_RATE = 0.02
+NUM_WARMUP_STEPS = 100
+WARMUP_PROPORTION = 0.0
+DROPOUT = 0.1
+# SHARED_SIZE = 512
+SHARED_SIZE = 0
+BATCH_SIZE = 256
```

Comment:
The batch size regulates how much of the training data is stored at once in memory. Dependent on the used machine 256 might be already too much.
Let's add this at least to the documentation...